src/app/reporting/Engine.cpp

```diff
@@ -350,7 +338,7 @@ CHIP_ERROR Engine::CheckAccessDeniedEventPaths(TLV::TLVWriter & aWriter, bool &
         {
             ReturnErrorOnFailure(err);
         }
-        else
+
```

Comment:
This change looks wrong to me.  This will encode status instead of the event if `err` is success, for example!  I _really_ hope this fails CI.

Reply:
This is indeed wrong, let me fix